docs/REST API/rest-api.md

```diff
@@ -3986,28 +3992,62 @@ List all configuration profiles for macOS hosts enrolled to Fleet's MDM that are
       "profile_id": 1337,
       "team_id": 0,
       "name": "Example profile",
+      "platform": "darwin",
       "identifier": "com.example.profile",
       "created_at": "2023-03-31T00:00:00Z",
       "updated_at": "2023-03-31T00:00:00Z",
       "checksum": "dGVzdAo="
+    },
+    {
+      "profile_id": 1338,
+      "team_id": 0,
+      "name": "Example Windows profile",
+      "platform": "windows",
+      "created_at": "2023-04-31T00:00:00Z",
+      "updated_at": "2023-04-31T00:00:00Z",
+      "checksum": "aCLemVr)"
     }
   ]
 }
 ```
 
-### Download custom macOS setting (configuration profile)
+### Get or download custom OS setting (configuration profile)
 
-`GET /api/v1/fleet/mdm/apple/profiles/{profile_id}`
+> [Download custom macOS setting](https://github.com/fleetdm/fleet/blob/ee02782eaf84c121256d73abc20b949d31bf2e57/docs/REST%20API/rest-api.md#download-custom-macos-setting-configuration-profile) (`GET /api/v1/fleet/mdm/apple/profiles/{profile_id}`) API endpoint is deprecated as of Fleet 4.40. It is maintained for backwards compatibility. Please use the API endpoint.
+
+`GET /api/v1/fleet/mdm/profiles/{profile_id}`
 
 #### Parameters
 
-| Name                      | Type    | In    | Description                                                               |
-| ------------------------- | ------- | ----- | ------------------------------------------------------------------------- |
-| profile_id                | integer | url   | **Required** The id of the profile to download.                           |
+| Name                      | Type    | In    | Description                                             |
+| ------------------------- | ------- | ----- | ------------------------------------------------------- |
+| profile_id                | integer | url   | **Required** The id of the profile to download.         |
+| alt                       | string  | query | If specified and set to "media", downloads the profile. |
```

Comment:
Do we use `alt` for scripts? Is this consistent with other API endpoints?

Reply:
More generally, why do we need the behavior in which `GET /profiles/{id}` returns information about the profile? Does the UI use this information?

Can we just have `GET /profiles/{id}` download the profile?

Reply:
> Do we use alt for scripts? Is this consistent with other API endpoints?

Yes, we use this approach for scripts already. See [here](https://fleetdm.com/docs/rest-api/rest-api#get-or-download-a-script).

>  Does the UI use this information?

No, we don't need this in UI right now. Same for scripts, but it's already implemented and might be useful at some point in the future. Here's some [discussion](https://github.com/fleetdm/fleet/issues/9829#issuecomment-1736138359) around `alt` for scripts API.

Should we keep it the same as for scripts and be consistent or remove it from profiles? What do you think @noahtalerman ?

Reply:
@mna We decided to match scripts, to have get or download (with `alt=media` param). This is specified in PR. For now it's just for profiles and we'll cover other endpoints later.